scripts/west_commands/runners/nrf_common.py

```diff
@@ -265,14 +265,14 @@ def program_hex(self):
             # This logic should be executed only once per build.
             # Use sysbuild board qualifiers to select the context, with which the artifacts will be programmed.
             if self.build_conf.get('CONFIG_BOARD_QUALIFIERS') == self.sysbuild_conf.get('SB_CONFIG_BOARD_QUALIFIERS'):
-                hex_path = Path(self.hex_)
+                mpi_hex_dir = Path(os.path.join(self.cfg.build_dir, 'zephyr'))
```

Comment:
nit: Mixing `pathlib` and `os.path` seems unnecessary
```suggestion
                mpi_hex_dir = Path(self.cfg.build_dir) / 'zephyr'
```